libs/metrics/src/launch_id.rs

```diff
@@ -0,0 +1,40 @@
+//! A type to uniquely identify a process.
+//! Useful to identify process restarts in logs and metrics.
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use prometheus::register_int_gauge_vec;
+
+pub struct LaunchId(String);
+
+impl LaunchId {
+    pub fn generate(prefix: &str) -> Self {
+        let startup_time_millis_since_epoch: u64 = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            .try_into()
+            .expect("many years to come until this doesn't fit into u63");
```

Comment:
Note we are not enforcing this to be u63 but u64, but I am unsure why would that be done?